bin/check-marathon-apps.rb

```diff
@@ -0,0 +1,383 @@
+    |    "deploying": {"status": 1},
+    |    "suspended": {"status": 1},
+    |    "waiting":   {"status": 1}
+    |  },
+    |  "health":{
+    |    "healthy":      {"status": 0},
+    |    "overcapacity": {"status": 1},
+    |    "staged":       {"status": 1},
+    |    "unhealthy":    {"status": 2},
+    |    "unscheduled":  {"status": 2},
+    |    "unknown":      {"status": 0}
+    |  }
+    |}
+  CONFIG
+
+  check_name 'CheckMarathonApps'
+  banner <<-BANNER.gsub(/^\s+\|/, '')
+    |Usage: ./check-marathon-apps.rb (options)
+    |
+    |This check will always return OK and publish two check results (health and status) per Marathon app.
+    |Marathon applications can override default_check_config by using labels as described below.
+    |
+    |Some example labels that can be used in Marathon application manifests:
+    |
+    |# Publish '"aggregate": "component"' field in the check results for all health and status checks.
+    |- SENSU_MARATHON_AGGREGATE=component
+    |# similar for some other fields
+    |- SENSU_MARATHON_CONTACT=support
+    |
+    |# Publish '"aggregate": "component"' field in the check results only for status checks.
+    |- SENSU_MARATHON_STATUS_AGGREGATE=component
+    |# Don't handle status check results
+    |- SENSU_MARATHON_STATUS_HANDLE=false
+    |
+    |# Generate UNKNOWN result for unknown health status, rather then the default OK result
+    |- SENSU_MARATHON_HEALTH_UNKNOWN_STATUS=3
+    |
+    |Similar logic could be applied for the provided default_check_config. Values
+    |under the special '_' key will be used as a default value for deeper levels.
+    |
+    |Here is the default value for default_check_config:
+    |#{DEFAULT_CHECK_CONFIG}
+    |
+    |Options:
+  BANNER
+
+  option :url,
+         description: 'Marathon API URL',
+         short: '-u url',
+         long: '--url url',
+         default: 'http://localhost:8080'
+
+  option :username,
+         short: '-u USERNAME',
+         long: '--username USERNAME',
+         default: '',
+         required: false,
+         description: 'Marathon API username'
+
+  option :password,
+         long: '--password PASSWORD',
+         default: '',
+         required: false,
+         description: 'Marathon API password'
+
+  option :match_pattern,
+         short: '-m PATTERN',
+         long: '--match-pattern PATTERN',
+         required: false,
+         description: 'Match app names against a pattern'
+
+  option :exclude_pattern,
+         short: '-x PATTERN',
+         long: '--exclude-pattern PATTERN',
+         required: false,
+         description: 'Exclude apps that match a pattern'
+
+  option :marathon_keys,
+         long: '--marathon-keys KEY1,KEY2,KEY3',
+         default: 'id,version,versionInfo,tasksStaged,tasksRunning,tasksHealthy,tasksUnhealthy,lastTaskFailure',
+         required: false,
+         description: 'Keys retrieved from Marathon API that will be included in the output'
+
+  option :default_check_config,
```

Comment:
I don't really know mesos would say more than 25% of users want to change this? Would this ever become a fairly large json blob or will it really only ever be a few objects? If we can pretty much always say it will remain fairly short that is fine but if it could make sense for it to be long I think specifying a file name is preferable.

I can understand not wanting to push that to CM, what about a compromise and say either pass in via arg or pass in a filename (separate option) which would take precedence over the file one specified here. I guess I see this the same as handlers typically needing a json config.

Reply:
I'm not sure if other users will need to modify, so followed your advice and added a flag to also support file config.

Reply:
I'd personally suggest the order of precedence be reversed and file should be the winner is specified. But that is moreso personal preference and I wont push if you feel otherwise. Also does default really make sense in the name if you are allowing to override it?